packages/solver-sat/src/solver-sat.ts

```diff
@@ -152,23 +152,28 @@ export async function solve(board: schema.Board, maxSolutions: number,
 
 function updateValidCandidatesForSolutions(
     solutions: IdxMap<typeof Geometry.CELL, number>[],
-    validCandidates: IdxMap<Geometry.CELL, Array<number>>,
+    validCandidates: IdxMap<Geometry.CELL, Map<number, number>>,
+	foundSolutions: Record<string, boolean>,
+	grid: Grid,
     size: number,
 ) {
     for (const solution of solutions) {
+		const representation = solutionToString(solution, grid);
+		if(representation in foundSolutions) continue;
```

Comment:
I think we typically put a space after an `if` (this comment applies to other places in your PR as well)